RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronCoreProducer.cc

```diff
@@ -0,0 +1,83 @@
+#include "DataFormats/EgammaCandidates/interface/GsfElectronCore.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronCoreFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/GsfPFRecTrack.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronCoreProducer.h"
+
+LowPtGsfElectronCoreProducer::LowPtGsfElectronCoreProducer( const edm::ParameterSet& config )
+  : GsfElectronCoreBaseProducer(config)
+{
+  superClusters_ = consumes<reco::SuperClusterCollection>(config.getParameter<edm::InputTag>("superClusters"));
+  superClusterRefs_ = consumes< edm::ValueMap<reco::SuperClusterRef> >(config.getParameter<edm::InputTag>("superClusters"));
+}
+
+LowPtGsfElectronCoreProducer::~LowPtGsfElectronCoreProducer() {}
+
+void LowPtGsfElectronCoreProducer::produce( edm::Event& event, const edm::EventSetup& setup ) {
+
+  // Output collection
+  auto electrons = std::make_unique<reco::GsfElectronCoreCollection>();
+
+  // Init
+  GsfElectronCoreBaseProducer::initEvent(event,setup) ;
+  if ( !useGsfPfRecTracks_ ) { edm::LogError("useGsfPfRecTracks_ is (redundantly) set to False!"); }
+  if ( !gsfPfRecTracksH_.isValid() ) { edm::LogError("gsfPfRecTracks handle is invalid!"); }
+  if ( !gsfTracksH_.isValid() ) { edm::LogError("gsfTracks handle is invalid!"); }
+
+  edm::Handle<reco::SuperClusterCollection> superClusters;
+  event.getByToken(superClusters_,superClusters);
+  if ( !superClusters.isValid() ) { edm::LogError("Problem with superClusters handle"); }
+
+  edm::Handle< edm::ValueMap<reco::SuperClusterRef> > superClusterRefs;
+  event.getByToken(superClusterRefs_,superClusterRefs);
+  if ( !superClusterRefs.isValid() ) { edm::LogError("Problem with superClusterRefs handle"); }
```

Comment:
Are these `LogError`s on invalid `edm::Handle`s really useful? Of these, `gsfPfRecTracksH_` and `superClusterRefs` will be accessed below so framework will throw an exception anyway. I didn't see `gsfTracksH_` or `superClusters` used at all, why why check them (and why read `superClusters` at all)?